flight/Modules/Battery/battery.c

```diff
@@ -227,8 +229,16 @@ static void batteryTask(void * parameters)
 			flightBatteryData.AvgCurrent = alpha * flightBatteryData.AvgCurrent + (1 - alpha) * flightBatteryData.Current; //in Amps
 
 			energyRemaining = batterySettings.Capacity - flightBatteryData.ConsumedEnergy; // in mAh
-			if (flightBatteryData.AvgCurrent > 0)
-				flightBatteryData.EstimatedFlightTime = (energyRemaining / (flightBatteryData.AvgCurrent * 1000.0f)) * 3600.0f; //in Sec
+
+			// And for time estimation, smooth things much more.
+			// Second order since it incorporates the above
+			// smoothing, 12s time constant for this layer.
+			alpha = 1.0f - dT / (dT + 12.0f);
+
+			time_lpf = time_lpf * alpha + (1 - alpha) * flightBatteryData.AvgCurrent;
```

Comment:
Not sure this is a particularly good variable name. `filtered_current` or something similar might be better?